### PR TITLE
9278 two clicks needed for show bailey

### LINKED ISSUE #9278: "Show Bailey" must be clicked twice
In https://habitica.com/user/settings/site , the button "Show Bailey" has to be clicked twice to actually show the Bailey. Bailey's most recent announcement should pop up after a single click.

### PR DESCRIPTION
**Pull request by perkyguy**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #9278 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
I updated how the notification handler and show bailey button in the settings page were handling the `newStuff` flag in `user.flags`. The `site.vue` and `newStuff.vue` were handling both showing and hiding the modal as well as setting the `newStuff` flag. This flag was also being   
 watched in `notifications.vue`, and showing the modal on any change. This caused the `modalStack` in `app.vue` to get full of `new-stuff` `modalID`, and it got very messy very quickly (sometimes two `show::modal` events would get emitted). The notification handler will still manage showing and hiding whenever the flag changes, however `site` and `newStuff` will only set/unset the flag if needed (on the "Show Bailey" button click, or "Dismiss this" button click) and hide the modal otherwise. 

[//]: # (Put User ID in here - found in Settings -> API)

----
UUID: e8be8049-ac14-401f-89ff-ce0aaa3445b4

### CODE REVIEW

File: website/client/components/notifications.vue [hunk 1]
```diff
@@ -210,7 +210,11 @@ export default {
   },
   watch: {
     baileyShouldShow () {
-      this.$root.$emit('show::modal', 'new-stuff');
+      if (this.user.flags.newStuff) {
+        this.$root.$emit('show::modal', 'new-stuff');
+      } else {
+        this.$root.$emit('hide::modal', 'new-stuff');
```
Comment (paglias) [inline website/client/components/notifications.vue]: does this get triggered every time the page loads?

**Comment by paglias** [comment]
Also can you fix the conflicts? Thanks!

File: website/client/components/notifications.vue (hunk 1 above)
Comment (perkyguy) [reply website/client/components/notifications.vue]: this only gets triggered on the flag change. On initial loading, the show/hiding of the modal is handled by `checkUserAchievements()`

**Comment by paglias** [comment]
@perkyguy I've just tested and the PR works well except for one small problem: if you show the Bailey modal and then refresh the page without dismissing it the modal seems to appear just to be hidden and then shown immediately after causing a weird animation to appear

**Comment by perkyguy** [comment]
I messed up my dev environment somehow, but I'll take a look at this as soon as I get it fixed

**Comment by paglias** [comment]
hey @perkyguy how are you doing on this?

**Comment by perkyguy** [comment]
Sorry, had a bunch of things going on, hadn't gotten back to this. I'll take care of this soon. I know I was just having some issues getting my dev environment to work correctly (all of the menus were broken). I'll get to this (and my other PRs) fixed and resubmitted soon!

**Comment by paglias** [comment]
@perkyguy I think this PR will be fixed by #9716 so I'm going to close it

**Pull request closed without merging**